libr/bin/p/bin_elf.inc

```diff
@@ -860,39 +854,38 @@ static void _patch_reloc(ut16 e_machine, RIOBind *iob, RBinElfReloc *rel, ut64 S
 }
 
 static RList* patch_relocs(RBin *b) {
+	r_return_val_if_fail (b, NULL);
 	RList *ret = NULL;
 	RBinReloc *ptr = NULL;
-	RIO *io = NULL;
-	RBinObject *obj = NULL;
-	struct Elf_(r_bin_elf_obj_t) *bin = NULL;
 	RIOMap *g = NULL;
 	HtUU *relocs_by_sym;
 	RBinElfReloc *relcs = NULL;
 	RBinInfo *info;
-	int cdsz;
-	int i;
-	ut64 n_vaddr, vaddr, size, offset = 0;
+	ut64 size, offset = 0;
 
-	if (!b)
+	RIO *io = b->iob.io;
+	if (!io || !io->desc) {
 		return NULL;
-	io = b->iob.io;
-	if (!io || !io->desc)
-		return NULL;
-	obj = r_bin_cur_object (b);
+	}
+	RBinObject *obj = r_bin_cur_object (b);
 	if (!obj) {
 	   	return NULL;
 	}
-	bin = obj->bin_obj;
+	struct Elf_(r_bin_elf_obj_t) *bin = obj->bin_obj;
 	if (bin->ehdr.e_type != ET_REL) {
 		return NULL;
 	}
+	size = bin->g_reloc_num * 4;
```

Comment:
why? also, size == 0 means g_reloc_num was 0 in the first place. Moreover, `g_reloc_num * 4` is already computed few lines below.

Reply:
Not exactly, int overflow can happen :}

```
$ cat a.c

int main() {
	short i = 16384;
	printf ("%hd\n", i * 4);
}
```

^- the output of this is `0`

Reply:
true, good point. we should define a macro or something in r_types_base.h to detect these kind of things

Reply:
however, if things overflow they can even become something different than 0, can't they? so i'm not sure size == 0 is the right check to check overflows.

Reply:
There's UT64_ADD_OVFCHK , but not for MUL, only for ADD

Reply:
Would something like https://stackoverflow.com/posts/1514309/revisions work?

Reply:
yep. these lines should be added as apis in r2 (or macros as UT64_MUL_OVFCHK)..

Reply:
Good, waiting for https://github.com/radareorg/radare2/pull/17216

Reply:
im fine to hold this merge until 17216 is in. but bin parsers are full of the need for this ovfchk, so maybe worth just address them together